server/tests/test_serializers.py

```diff
@@ -1,17 +1,1421 @@
 import unittest
 import logging
-from cjwkernel.types import I18nMessage, QuickFix, QuickFixAction, RenderError
-from server.serializers import jsonize_i18n_message, JsonizeContext
-from cjworkbench.i18n.trans import MessageTranslator
+from cjwkernel.types import (
+    I18nMessage,
+    I18nMessageSource,
+    QuickFix,
+    QuickFixAction,
+    RenderError,
+)
+from server.serializers import (
+    jsonize_i18n_message,
+    JsonizeContext,
+    JsonizeModuleContext,
+    jsonize_clientside_module,
+    _jsonize_param_spec,
+)
+from cjworkbench.i18n.trans import MessageLocalizer
 from unittest.mock import patch
 from babel.messages.catalog import Catalog
+from cjwstate.modules.types import ModuleSpec
+from cjwstate.clientside import Module
+from typing import Dict, Any
 
 
-def mock_jsonize_context(user=None, session=None, locale_id=None):
-    return JsonizeContext(user=user, session=session, locale_id=locale_id)
+def mock_jsonize_context(
+    user=None, session=None, locale_id=None, localizers={}, module_id=None
+):
+    if module_id:
+        return JsonizeModuleContext(
+            user=user,
+            session=session,
+            locale_id=locale_id,
+            module_id=module_id,
+            localizers=localizers,
+        )
+    else:
+        return JsonizeContext(
+            user=user, session=session, locale_id=locale_id, localizers=localizers
+        )
+
+
+def _jsonize_clientside_module_return(args: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
This should be a variable, not a function. Call it `DEFAULT_SERIALIZED_MODULE = { ... }`

And then callers should compare with `{**DEFAULT_SERIALIZED_MODULE, ...}`